sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -204,6 +204,13 @@ object SQLConf {
     .intConf
     .createWithDefault(4)
 
+  val LIMIT_FLAT_GLOBAL_LIMIT = buildConf("spark.sql.limit.flatGlobalLimit")
+    .internal()
+    .doc("During global limit, try to evenly distribute limited rows across data " +
+      "partitions. If disabled, scanning data partitions sequentially until reaching limit number.")
+    .booleanConf
+    .createWithDefault(true)
```

Comment:
I set this as true. One reason is to see if it can pass existing tests. If we don't feel confident or worry about behavior change, we can set this to false before merging.

Reply:
@hvanhovell Should we set it false by default?

Reply:
so i read this config doc five times, and i still couldn't figure out what it does, until i went ahead and read the implementation.